packages/components/src/styles/componentTypes.ts

```diff
@@ -1,48 +1,93 @@
-export type TableStyles = {
-    root: string
-    table: string
-    header: string
-    headerRow: string
-    headerCell: string
-    body: string
-    cell: string
-    row: string
-    selectedRow: string
-    subRow: string
-  }
+export interface BaseStyles {
+  root: React.HTMLAttributes<HTMLDivElement>['className'];
+}
+
+export interface TableStyles extends BaseStyles {
+  table: string;
+  header: string;
+  headerRow: string;
+  headerCell: string;
+  body: string;
+  cell: string;
+  row: string;
+  selectedRow: string;
+  subRow: string;
+}
 
 type StyleVariant = {
-    solid: string,
-    outlined: string
-}
-
-export type ButtonStyles = {
-    root: string,
-    primary: StyleVariant,
-    secondary: StyleVariant,
-    neutral: StyleVariant,
-    sm: string,
-    md: string,
-    lg: string,
-    fullWidth: string,
-    rounded: string,
-    highlightOnHover: string,
-    icon: string,
-    disabled:  string,
-    button: string,
-  }
-
-export type IconStyles = {
-    root: string,
-    primary: string,
-    secondary: string,
-    neutral: string,
-    sm: string,
-    md: string,
-    lg: string,
-    muted: string,
-    rounded: string,
-    highlightOnHover: string,
-    brightenOnHover: string,
-    bright:  string,
-  }
+  solid: string;
+  outlined: string;
+};
+
+export interface ButtonStyles extends BaseStyles {
+  primary: StyleVariant;
+  secondary: StyleVariant;
+  neutral: StyleVariant;
+  sm: string;
+  md: string;
+  lg: string;
+  fullWidth: string;
+  rounded: string;
+  highlightOnHover: string;
+  icon: string;
+  disabled: string;
+  button: string;
+}
+
+export interface IconStyles extends BaseStyles {
+  primary: string;
+  secondary: string;
+  neutral: string;
+  sm: string;
+  md: string;
+  lg: string;
+  muted: string;
+  rounded: string;
+  highlightOnHover: string;
+  brightenOnHover: string;
+  bright: string;
+}
+
+export interface FieldSetStyles extends BaseStyles {
+  label: React.HTMLAttributes<HTMLDivElement>['className'];
```

Comment:
can we create type as `ClassName = React.HTMLAttributes<HTMLDivElement>['className']` and use it in all ocurrencies? I think it would be cleaner to read, just a preference.